tools/generator/xcodeproj_targets.bzl

```diff
@@ -1,6 +1,24 @@
 """Exposes targets used by `xcodeproj` to allow use in fixture tests."""
 
-XCODEPROJ_TARGETS = [
-    "//tools/generator:generator",
-    "//tools/generator/test:tests",
-]
+load("@bazel_skylib//lib:sets.bzl", "sets")
+load("//xcodeproj:xcodeproj.bzl", "xcode_schemes")
+
+def get_xcode_schemes():
```

Comment:
Does this have to be a function versus a constant?

Reply:
Unfortunately, yes. If I do the following in the file,

```python
_SCHEMES = [
    xcode_schemes.scheme(
        name = "Generator",
        launch_action = xcode_schemes.launch_action(
            "//tools/generator:generator",
        ),
        test_action = xcode_schemes.test_action([
            "//tools/generator/test:tests",
        ]),
    ),
]
```

we experience the thread error that I mention in the description:

```
Error in repository_name: The native module can be accessed only from a BUILD thread. Wrap the function in a macro and call it from a BUILD file
```

To preserve the sharing of the targets list, I wrapped them in the functions that you see. They are called on the correct on the correct thread.